src/boot/AppEventHandlers.js

```diff
@@ -95,6 +96,17 @@ class AppEventHandlers extends PureComponent<Props> {
     if (state === 'background' && Platform.OS === 'android') {
       NativeModules.BadgeCountUpdaterModule.setBadgeCount(unreadCount);
     }
+    if (state === 'active' && Platform.OS === 'ios') {
+      // TODO: This API is deprecated in react-native-notifications release
+      // 2.0.6-snapshot.8; see #3647.
+      try {
```

Comment:
See motivation for try/catch at https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/iOS.20push.20notifications/near/825135.

Reply:
I dug into this a bit – but, unfortunately, not enough to have a solid answer to that question.

I have learned that it probably *shouldn't* throw. The [definition of `consumeBackgroundQueue`](https://github.com/wix/react-native-notifications/blob/v1.4.0/RNNotifications/RNNotifications.m#L549) uses [`RCT_EXPORT_METHOD`](https://github.com/facebook/react-native/blob/v0.59.10/React/Base/RCTBridgeModule.h#L134-L175), which creates [an asynchronous method](https://github.com/facebook/react-native/blob/v0.59.10/React/Base/RCTBridgeModule.h#L208-L210) which [doesn't return any sort of value](https://github.com/facebook/react-native/blob/v0.59.10/React/Base/RCTModuleMethod.mm#L557-L562). (Even via callbacks, since it has no parameters.)

However, the invocation of a native method is eventually done [here](https://github.com/facebook/react-native/blob/v0.59.10/React/CxxModule/RCTNativeModule.mm#L91-L120), and the comment [here](https://github.com/facebook/react-native/blob/v0.59.10/React/CxxModule/RCTNativeModule.mm#L108) implies that some `NSException`s are propagated back to JS. I haven't found where that would happen.